src/utils/auth/guestCourseEnrollment.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * Utilities for managing guest user course enrollment
+ * Guest enrollments are stored in localStorage to track which courses
+ * a guest user has "enrolled" in for courses with allowGuestAccess=true
+ */
+
+import { logErrorToSentry } from '@/lib/sentry';
+import { isLoggedIn } from '@/utils/auth/login';
+
+const GUEST_ENROLLED_COURSES_KEY = 'guestEnrolledCourses';
+
+export type UserType = 'logged_in' | 'guest';
+
+/**
+ * Get all course IDs that the guest user has enrolled in (internal use only)
+ *
+ * @returns {string[]} An array of course IDs that the guest user has enrolled in
+ *
+ */
+const getGuestEnrolledCourses = (): string[] => {
```

Comment:
Is there a reason why we are using local stroage vs redux toolkit slice like the rest of the app?